src/lib/cache.js

```diff
@@ -21,12 +22,14 @@ const VerboseEvents = ["issue", "failure", "reconnecting", "reconnect", "remove"
 
 const deflateP = (dataz) => {
   return new Promise((resolve, reject) =>
-    zlib.deflate(JSON.stringify(dataz), (er, deflatedData) => {
-      if (er) {
-        error(er)
-      } else {
-        resolve(deflatedData)
-      }
+    yj.stringifyAsync(dataz, (err, data) => {
+      zlib.deflate(JSON.stringify(dataz), (er, deflatedData) => {
```

Comment:
If we are actually CPU-bound, is it possible that compression could make things worse?

Reply:
Hmm, I see from the article in the PR description that `zlib` runs on a worker thread. So maybe not an issue from an event loop blocking perspective, but still, would add load if our worker threads are also CPU-bound.

Reply:
That load has to happen somewhere, though, right? Moving it to workers can only help, I think?

Reply:
Sorry, this wasn't pertaining to using `yj`, but to the recently introduced compression.

Reply:
ooooh, I see what you mean. That's a good point, if we're CPU bound compression would make it worse not better!